builtin/game/chatcommands.lua

```diff
@@ -815,12 +815,18 @@ core.register_chatcommand("days", {
 })
 
 core.register_chatcommand("shutdown", {
+    params = "<delay_in_seconds(0..inf)> | Leave blank for normal shutdown",
```

Comment:
@nerzhul That kind of thing is not possible with Lua patterns. It's not a full-fledged RE engine.

It's not worth the effort to restrict anything as long as there's some feedback noting the computed value, but if it's really important, here's a somewhat better implementation:

```lua

local unit_to_secs = {
	s = 1, m = 60, h = 3600,
	D = 86400, W = 604800, M = 2592000, Y = 31104000,
}

local function parse_time(t)
	local tm = { }
	for k in pairs(unit_to_secs) do
		tm[k] = 0
	end
	for num, unit in t:gmatch("(%d+)([smhDWMY]?)") do
		if unit == "" then unit = "s" end
		tm[unit] = tonumber(num)
	end
	local total = 0
	for k, v in pairs(tm) do
		total = total + unit_to_secs[k]*v
	end
	return total
end

assert(parse_time("1m2m") == 120)
```